assets/src/pages/types.ts

```diff
@@ -14,3 +14,38 @@ export const isTrackType = (value: string): value is TrackType => TrackTypeValue
 
 const StreamSourceValues = ["local", "remote"] as const;
 export type StreamSource = (typeof StreamSourceValues)[number];
+
+export type TrackWithId = {
+  stream?: MediaStream;
+  remoteTrackId: string | null;
+  encodingQuality?: TrackEncoding;
+  metadata?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+  enabled?: boolean;
+};
+
+// Media Tile Types
+type CommonTile = {
+  mediaPlayerId: string;
+  peerId: string;
+  video: TrackWithId | null;
+  displayName: string;
+  streamSource: StreamSource;
+};
+
+export type LocalTileConfig = {
+  typeName: "local";
+  audio: TrackWithId | null;
+  initials: string;
+} & CommonTile;
+
+export type RemoteTileConfig = {
+  typeName: "remote";
+  audio: TrackWithId | null;
+  initials: string;
+} & CommonTile;
```

Comment:
Can we avoid this repetition? You could extract common props to a separate type or create a union type for typeName: `"local" | "remote"`